package/MDAnalysis/guesser/default_guesser.py

```diff
@@ -0,0 +1,427 @@
+# -*- Mode: python; tab-width: 4; indent-tabs-mode:nil; coding: utf-8 -*-
+# vim: tabstop=4 expandtab shiftwidth=4 softtabstop=4
+#
+# MDAnalysis --- https://www.mdanalysis.org
+# Copyright (c) 2006-2017 The MDAnalysis Development Team and contributors
+# (see the file AUTHORS for the full list of names)
+#
+# Released under the GNU Public Licence, v2 or any higher version
+#
+# Please cite your use of MDAnalysis in published work:
+#
+# R. J. Gowers, M. Linke, J. Barnoud, T. J. E. Reddy, M. N. Melo, S. L. Seyler,
+# D. L. Dotson, J. Domanski, S. Buchoux, I. M. Kenney, and O. Beckstein.
+# MDAnalysis: A Python package for the rapid analysis of molecular dynamics
+# simulations. In S. Benthall and S. Rostrup editors, Proceedings of the 15th
+# Python in Science Conference, pages 102-109, Austin, TX, 2016. SciPy.
+# doi: 10.25080/majora-629e541a-00e
+#
+# N. Michaud-Agrawal, E. J. Denning, T. B. Woolf, and O. Beckstein.
+# MDAnalysis: A Toolkit for the Analysis of Molecular Dynamics Simulations.
+# J. Comput. Chem. 32 (2011), 2319--2327, doi:10.1002/jcc.21787
+
+from .base import GuesserBase
+import numpy as np
+import warnings
+
+import re
+
+from ..lib import distances
+from . import tables
+
+
+class DefaultGuesser(GuesserBase):
+    context = 'default'
+
+    def __init__(self, universe, **kwargs):
+        super().__init__(universe, **kwargs)
+        self._guess = {'masses': self.guess_masses,
+                       'types': self.guess_types,
+                       'angles': self.guess_angles,
+                       'dihedrals': self.guess_dihedrals,
+                       'bonds': self.guess_bonds}
+
+    def guess_masses(self):
+        """Guess the mass of many atoms based upon their type
+
+        Returns
+        -------
+        atom_masses : np.ndarray dtype float64
+        """
```

Comment:
Awesome work on this PR! Many topology files include types but not masses, which `guess_masses` is perfect for, but there is at least one, LAMMPS `.data` files that includes masses but not types or elements (see issue #3394). May be worth considering a `guess_elements` function in `default_guesser.py` that can infer elements from masses.

Reply:
I have just seen this review now sorry for that! it's a great point that I really would have messed if you didn't point me out to it, thank you!